### ISSUE TITLE
kubectl resources autocompletion

### ISSUE THREAD

**Issue by giolf** [warpdotdev/Warp#2055]
### Describe the solution you'd like?
Currently warp allows to autocomplete kubectl commands like:
`kubectl get` or `kubectl describe`

However it doesn't provide resources autocompletion, for example:
if you type: `kubectl describe pod -n my-namespace` and then if you press `TAB` it doesn't show you the list of the pods in that particular namespace. 

I tried to setup on my `.zshrc`file the following instruction:
```
# kubectl config
source <(kubectl completion zsh)
```
and on iTerm2 it works, but not in warp. 

I think it would be super cool to provide this extra layer of autocompletion (similarly to the autocompletion warp offer with git)

### How important is this feature to you?

4

**Comment by dannyneira** [comment]
Thanks for this feature request! Please +1 this request as it helps us gauge interest. We will post any updates here.

**Comment by pierre-mike-pge** [comment]
+1

**Comment by cedi** [comment]
+1

**Comment by giolf** [comment]
+1

**Comment by macro-dadt** [comment]
+1

**Comment by muneebshahid** [comment]
+1

**Comment by pavan2004it** [comment]
+1

**Comment by piavgh** [comment]
+1

**Comment by piavgh** [comment]
After having this feature, I will remove iTerm2 😄

**Comment by lummie** [comment]
+1

**Comment by RapidIE** [comment]
+1 
+10
+1000 =)

**Comment by unbreakabl3** [comment]
+1 for that functionality

**Comment by vincent-at-trust** [comment]
+1

**Comment by lkoniecz** [comment]
+1

**Comment by pavan2004it** [comment]
+1

**Comment by gomgomshrimp** [comment]
+1 
This feature is very important. 
People who often work with Kubernetes avoid warp because of this feature.

**Comment by Elias-Chairi** [comment]
+1

**Comment by ovstuckrad** [comment]
Definitely need this to use Warp!

**Comment by ErlendFax** [comment]
+1

**Comment by clementblaise** [comment]
+1

**Comment by MiddleCloud** [comment]
+1

**Comment by domechn** [comment]
+1

**Comment by hreinking** [comment]
+1, a deal-breaker for me, and loving Warp so far.

**Comment by nianyush** [comment]
+1

**Comment by ebenezar-mccoy** [comment]
+1

**Comment by rayanebel** [comment]
+1

**Comment by fatalwithin** [comment]
+1

**Comment by jentyk** [comment]
+1

IMHO I would not say it is a new feature. This is a major regressions comparing to the original functionality.

**Comment by bygui86** [comment]
FYI I tried also `gcloud` and, same as kubectl, only for "first portion" of command autocompletion works...

**Comment by DTcloudOps** [comment]
+1 plz

**Comment by 808Mak1r** [comment]
+1

**Comment by liaddrori1** [comment]
+1

**Comment by bad33ndj3** [comment]
+1

**Comment by benpsnyder** [comment]
Pleeeeeease :)

**Comment by smorenburg** [comment]
As already mentioned, this is a deal breaker. I've just switched back to iTerm until this is resolved.
I love what you're doing, though!

**Comment by JeremyCurmi** [comment]
same here +1, really need this feature. Great work warp :)

**Comment by anoopwebs** [comment]
Warp already made a significant difference but I'd love to see Warp have this feature too.

**Comment by mar-pan** [comment]
Hello any chance that this feature could be available anytime soon?

**Comment by humw** [comment]
+1

**Comment by patrickschaffrath** [comment]
+1

**Comment by aranya-danno** [comment]
+1

**Comment by jbnjohnathan** [comment]
+1

**Comment by zcl0621** [comment]
+1 pls

**Comment by dannyneira** [comment]
Hey Folks, this is on our radar to fix/improve, no ETA on the launch but it's something we're looking into so we appreciate your patience. We'll post any more updates on this thread.

**Comment by antonkobylko1990** [comment]
+1!

**Comment by petrvnk** [comment]
+1

**Comment by aviadhaham** [comment]
Wow, amen to all mentioned above; I mean, only after switching from iTerm2 I noticed that, and that's a HUGE one.
Please prioritize it :(

**Comment by youngcm2** [comment]
+1

**Comment by kelt1k** [comment]
+100

**Comment by taras-klym** [comment]
+1

**Comment by Roman-Axilis** [comment]
+1

**Comment by tomicakr** [comment]
+1

**Comment by An0nymous0** [comment]
+1

**Comment by Owenxh** [comment]
+10086

**Comment by vlad-artyomov** [comment]
+1

**Comment by Artebomba** [comment]
+1

**Comment by RomanGrekov** [comment]
+1

**Comment by dimm0** [comment]
Any estimate? I'm almost ready to switch back to iTerm, as there's no progress since November...

**Comment by paolos7709** [comment]
> Any estimate? I'm almost ready to switch back to iTerm, as there's no progress since November...

Same here and I already had to. For me it is daily driver...

**Comment by ebenezar-mccoy** [comment]
The best decision I made was to start using the built-in terminal in VS Code. Inability to have "native" integration with the editor, non-working autocompletion, are reasons that make it a no-go for me.

**Comment by maxheyer** [comment]
+1

**Comment by levya4** [comment]
+1000

**Comment by igorcezar** [comment]
+1

**Comment by ShonL** [comment]
+1
Going back to iTerm unfortunately

**Comment by kelt1k** [comment]
![image](https://github.com/warpdotdev/Warp/assets/62957498/f4cd9842-57ac-40da-b3a8-ecbb63444f68)
thanks guys ) it's very useful! (really nope)

**Comment by mschaefer-gresham** [comment]
I might add that improving Warps default prompt to show k8s namespace/context information is just as important. Please add both features at the same time.

**Comment by dannyneira** [comment]
Hey Folks, this is something on our radar as there is room for improvement in k8s support. No official ETA, but we'll post any updates on this thread when we have them.

**Comment by mohanraj-nagasamy** [comment]
+1 for this feature.

**Comment by jasine** [comment]
+1

**Comment by devav2** [comment]
+1 Will be nice to have that feature rolled out quickly

**Comment by schabrolles** [comment]
+1 for me

**Comment by imans777** [comment]
+1

**Comment by JonasHess** [comment]
+1

**Comment by lieberlois** [comment]
+1

**Comment by valdirjunior011** [comment]
+1

**Comment by jzuijlek** [comment]
+1

**Comment by noahzweben** [comment]
Hi all - I'm Noah, a PM at Warp. We finally have a Warp demo build that has support for kubernetes completions! 

Since we don't use kubernetes at Warp we're looking for people who want to test this in the wild! **If you're interested in trialing kubernetes completions before we publicly release it,  please email me at **noah@warp.dev****

It adds support for suggesting:
* k8s resources
* global options (e.g. kubectl options)
* when --namespace and/or --kubeconfig are supplied, scope down suggestions to the specified namespace/kubeconfig

**Comment by that70schris** [comment]
is it possible to add support for `kubecolor` as well?

**Comment by dannyneira** [comment]
Good news folks! The [latest version of Warp](https://docs.warp.dev/getting-started/changelog#2023.11.30-v0.2023.11.28.08.02) now supports additional `kubectl` completions, including suggestions for resources, global options, and namespaces!

![CleanShot 2023-12-01 at 09 29 51](https://github.com/warpdotdev/Warp/assets/16809145/3412ee35-c7f9-4ff7-b07e-2f98e89cec99)

Closing as completed, but please feel free to let us know if any bugs by opening a [new GitHub issue](https://github.com/warpdotdev/Warp/issues/new/choose).

**Comment by dannyneira** [comment]
Please note the following mentioned in a [discord thread](https://discord.com/channels/851854972600451112/1180130732748841041):
> Hi! I usually use Starship to display the current K8s context. I tried the new Warp prompt chip, but it doesn't seem to work.
Was anyone able to get it running?
> I was able to get it to work once I moved all my contexts into a single config file at ~/.kube/config. The prompt chip doesn't seem to honor the KUBECONFIG env var

**Comment by schabrolles** [comment]
Yes, that's a good step. Thank hope to have openshift cli soon (oc) which is based on kubectl.

**Comment by imans777** [comment]
> Cool feature！！！Have been used for several days, but there is still some suggestions,only typing kubectl there is k8s-context showing in prompt,i think it can be shown all the time,or type k,because we all alias kubectl as k :( Thanks :)

You can use a workaround here, there's a feature in Warp: "Expand aliases as you type". when you enable this, whenever you type `k` it automagically converts it to `kubectl` and then you can have all kubectl autocompletion active! This is also useful for other commands or aliases that you've set.

**Comment by AdamGold** [comment]
Hey, there's a problem with auto completing `kubectl logs` -- it doesn't autocomplete the resource name until you do `kubectl logs pod .....` but that's not how kubectl command works with logs, it expects the resource name right after `kubectl logs`.

**Comment by rasifdo** [comment]
finally one of the most awaited features! hoping the team fixes autocompletion for 'k logs' as @AdamGold  mentioned

**Comment by dannyneira** [comment]
@lkoniecz [Unfortunately not](https://docs.warp.dev/appearance/prompt#kubernetes) but this may be improved upon in the future, you'd have to wrap the alias in a conditional to disable it in Warp for now.

```
# Bash and Zsh
if [[ $TERM_PROGRAM != "WarpTerminal" ]]; then
# > What you want to disable here <
  alias k='/opt/homebrew/bin/kubectl'
fi

# Fish
if test "$TERM_PROGRAM" != "WarpTerminal"
# > What you want to disable here <
  alias k='/opt/homebrew/bin/kubectl'
end
```

**Comment by lkoniecz** [comment]
@dannyneira I don;t think I need to disable that - it still works with `kubectl`.

**Comment by imans777** [comment]
just to make sure this is not missed, I'm copying my comment about my experience on kubectl missing autocompletions here too:
https://github.com/warpdotdev/Warp/discussions/652#discussioncomment-7742760

**Comment by nizametdinovcrx** [comment]
> Hey, there's a problem with auto completing `kubectl logs` -- it doesn't autocomplete the resource name until you do `kubectl logs pod .....` but that's not how kubectl command works with logs, it expects the resource name right after `kubectl logs`.

Is this resolved?

**Comment by zcl0621** [comment]
@nizametdinovcrx  also not support...

**Comment by cx2c** [comment]
I need type external `pod`  to  complet pod name. but `kubectl logs -n xxx pod xxxx` is not a kubectl command
<img width="715" alt="image" src="https://github.com/warpdotdev/Warp/assets/13431516/9adfc47e-336a-43c2-8116-e03faedd5e85">

**Comment by yura-kutsiy** [comment]
Join the open issue (related kubectl log autocompletion feature) and leave your thumbs up.

#3958

**Comment by rubengmurray** [comment]
Doesn't work for me at all in conjuction with ohmyzsh

**Comment by mgmoghaddam** [comment]
need this future , kubectl-autocomplete not working on warp over zsh ohmyzsh

**Comment by kopaygorodsky** [comment]
`kubectl logs` still does not work as some other items. Would be nice to be able to disable autocomplete for certain binaries.

**Comment by karimeid-is** [comment]
any updates on the resource name auto completion part, this is the only feature preventing me from never using terminal again. Currently needing to use an extra get pod, service... command just to get the pod name to be used in the log or describe command is inconvenient.

**Comment by rubengmurray** [comment]
This needs to happen. It's my only complaint with warp.

**Comment by mschaefer-gresham** [comment]
Please add this feature and feature request [3978](https://github.com/warpdotdev/Warp/issues/3978) at the same time please.

**Comment by Honigeintopf** [comment]
Is this implemented, if yes how do I activate it? Right now it seems to not work.
